internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -496,11 +497,11 @@ private Future<RecordMetadata> delete(
       Callback callback,
       LeaderMetadataWrapper leaderMetadataWrapper,
       long logicalTs,
-      Optional<DeleteMetadata> deleteMetadata) {
+      DeleteMetadata deleteMetadata) {
     byte[] serializedKey = keySerializer.serialize(topicName, key);
     isChunkingFlagInvoked = true;
 
-    int rmdPayloadSize = deleteMetadata.map(DeleteMetadata::getSerializedSize).orElse(0);
+    int rmdPayloadSize = deleteMetadata == null ? 0 : deleteMetadata.getSerializedSize();
```

Comment:
this just reads so much better.  How the heck did we ever land on optionals????

Reply:
🤷 🙁